### PR TITLE
chore: add `fedimint-core` high-level docstrings

### PR DESCRIPTION
**Pull request by dpc**
Please don't be too nit-picky. It's better than nothing.

### CODE REVIEW

File: fedimint-core/src/lib.rs [hunk 1]
```diff
@@ -24,33 +42,56 @@ pub use crate::core::server;
...
+/// Types handling multiple per-denomination values
 pub mod tiered_multi;
+/// Time handling, wasm safe functionality
 pub mod time;
+/// Timing helpers
 pub mod timing;
+/// Fedimint transaction (inpus + outputs + signature) types
 pub mod transaction;
+/// Peg-in txo proofs
 pub mod txoproof;
```
Comment (m1sterc001guy) [inline fedimint-core/src/lib.rs:93]: not blocking, but maybe this should move out of `fedimint-core` at some point?

**Review by m1sterc001guy — APPROVED** [review]

File: fedimint-core/src/lib.rs (hunk 1 above)
Comment (dpc) [reply fedimint-core/src/lib.rs:93]: Added. https://github.com/fedimint/fedimint/issues/4691

File: fedimint-core/src/encoding/mod.rs [hunk 2]
```diff
@@ -1,6 +1,10 @@
 //! This module defines a binary encoding interface which is more suitable for
 //! consensus critical encoding than e.g. `bincode`. Over time all structs that
 //! need to be encoded to binary will be migrated to this interface.
+//!
+//! This code is based on corresponding `rust-bitcoin` types.
```
Comment (elsirion) [inline fedimint-core/src/encoding/mod.rs:5]: `s/types/traits/`?

File: fedimint-core/src/lib.rs [hunk 3]
```diff
@@ -24,33 +42,56 @@ pub use crate::core::server;
...
 pub mod envs;
 pub mod epoch;
+/// Formatting helpers
 pub mod fmt_utils;
+/// Hex encoding helpers
 pub mod hex;
+/// Common macros
 #[macro_use]
 pub mod macros;
+/// Extenable module sysystem
```
Comment (elsirion) [inline fedimint-core/src/lib.rs:74]: Extendable?

File: fedimint-core/src/lib.rs [hunk 4]
```diff
@@ -24,33 +42,56 @@ pub use crate::core::server;
...
+/// Peer networking
 pub mod net;
+/// Client query system
 pub mod query;
+/// Task handling, including wasm safe logic
 pub mod task;
+/// Types handling per-denomination values
 pub mod tiered;
+/// Types handling multiple per-denomination values
 pub mod tiered_multi;
```
Comment (elsirion) [inline fedimint-core/src/lib.rs:85]: Should probably move into mint-core

**Review by elsirion — APPROVED** [review]

**Pull request merged**